app/views/csvfiles/new.html.erb

```diff
@@ -0,0 +1,155 @@
+                let name = data[0]["csvfile"]["uid"] + "user_id";
+                let index = $('input[name='+ name +']:checked').val();
+                let allfiles = JSON.parse(data[index]["csvfile"]["filepath"]);
+                SimpleDataGrapherObject.view.usingPreviouslyUploadedFile();
+                SimpleDataGrapherObject.view.csvParser.completeCsvMatrix = allfiles["completeCsvMatrix"];
+                SimpleDataGrapherObject.view.csvParser.csvHeaders = allfiles["csvHeaders"];
+                SimpleDataGrapherObject.view.csvParser.csvSampleData = allfiles["csvSampleData"];
+                SimpleDataGrapherObject.view.csvParser.csvValidForYAxis = allfiles["csvValidForYAxis"];
+                SimpleDataGrapherObject.view.csvParser.completeCsvMatrixTranspose = allfiles["completeCsvMatrixTranspose"];
+                SimpleDataGrapherObject.view.continueViewManipulation("prevfile");
+            }); 
+        }
+        $("#" + SimpleDataGrapherObject.view.elementId + "_publish").click(function(){
+            console.log(SimpleDataGrapherObject.view.plotlyjsPlotter.dataHash);
+            console.log(SimpleDataGrapherObject.view.plotlyjsPlotter.graphType);
+            var arr = {};
+            arr["completeCsvMatrix"] = SimpleDataGrapherObject.view.csvParser.completeCsvMatrix;
+            arr["csvHeaders"] = SimpleDataGrapherObject.view.csvParser.csvHeaders;
+            arr["csvSampleData"] = SimpleDataGrapherObject.view.csvParser.csvSampleData;
+            arr["csvValidForYAxis"] = SimpleDataGrapherObject.view.csvParser.csvValidForYAxis;
+            arr["completeCsvMatrixTranspose"] = SimpleDataGrapherObject.view.csvParser.completeCsvMatrixTranspose;
+            let csvStringMatrix = SimpleDataGrapherObject.view.csvParser.completeCsvMatrixTranspose;
+            let csvStringForDownload = encodeURI("data:text/csv;charset=utf-8," + csvStringMatrix.map(e => e.join(",")).join("\n"));
+            var dataObject = {};
+            dataObject["hash"] = SimpleDataGrapherObject.view.plotlyjsPlotter.dataHash;
+            dataObject["graphType"] = SimpleDataGrapherObject.view.plotlyjsPlotter.graphType;
+            dataObject["length"] = SimpleDataGrapherObject.view.plotlyjsPlotter.length;
+            $.ajax({
+                url: '/graph/note/graphobject',
+                type: 'post',
+                data: {object: JSON.stringify(arr), 
+                        uid: <%= current_user.id %>, 
+                        filetitle: SimpleDataGrapherObject.view.fileTitle, 
+                        filedescription: SimpleDataGrapherObject.view.fileDescription, 
+                        filestring: csvStringForDownload,
+                        graphobject: JSON.stringify(dataObject)},
+                success: function(data){
+                    console.log("saved",data);
+                    window.location = '/post?body=%5Bsimple-data-grapher:i/'+data["uid"]+'/'+data["id"]+'%5D';
```

Comment:
I am little doubtful about this `window.location` statement. Can you explain once about what are you trying to achieve here. 
To me it looks like a `GET` request with params being sent. If that is really the case then maybe this needs a server side fix of redirecting the user on successful request. It can be discussed later on, I would first like to know your view of this line.

Reply:
This is actually an integration between simple-data-grapher to the editor controller, and I have done it in the manner discussed with @jywarren, so that none of the existing controller or view code gets disturbed by this feature. So this shouldn't be a problem, I will take care of the remaining reviews you have posted, thanks a lot!!

Reply:
Can you provide a little detail on that or point me to the discussion? I am concerned about hardcoding the url with unicode characters.

Reply:
Actually, the hard-coding is just so that the text appears in `Power tag` body, those are just to style it, rest are general parameters.

Reply:
I can't say anything for now since I don't have a deeper insight on it. Let's keep it that way for now. Maybe later it can be made better if possible.